zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/authorization/DeleteRoleMultiPartitionTest.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.engine.processing.authorization;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.AssertionsForClassTypes.tuple;
+
+import io.camunda.zeebe.engine.state.distribution.DistributionQueue;
+import io.camunda.zeebe.engine.util.EngineRule;
+import io.camunda.zeebe.protocol.record.Record;
+import io.camunda.zeebe.protocol.record.RecordType;
+import io.camunda.zeebe.protocol.record.ValueType;
+import io.camunda.zeebe.protocol.record.intent.AuthorizationIntent;
+import io.camunda.zeebe.protocol.record.intent.CommandDistributionIntent;
+import io.camunda.zeebe.protocol.record.intent.RoleIntent;
+import io.camunda.zeebe.protocol.record.value.AuthorizationOwnerType;
+import io.camunda.zeebe.protocol.record.value.AuthorizationResourceType;
+import io.camunda.zeebe.protocol.record.value.CommandDistributionRecordValue;
+import io.camunda.zeebe.protocol.record.value.PermissionAction;
+import io.camunda.zeebe.test.util.record.RecordingExporter;
+import io.camunda.zeebe.test.util.record.RecordingExporterTestWatcher;
+import java.time.Duration;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Collectors;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestWatcher;
+
+public class DeleteRoleMultiPartitionTest {
+
+  private static final int PARTITION_COUNT = 3;
+
+  @Rule public final EngineRule engine = EngineRule.multiplePartition(PARTITION_COUNT);
+
+  @Rule public final TestWatcher testWatcher = new RecordingExporterTestWatcher();
+
+  @Test
+  public void shouldDistributeRoleDeleteCommand() {
+    // when
+    final var name = UUID.randomUUID().toString();
+    final var roleKey = engine.role().newRole(name).create().getValue().getRoleKey();
+    engine
+        .authorization()
+        .permission()
+        .withOwnerKey(roleKey)
+        .withOwnerType(AuthorizationOwnerType.ROLE)
+        .withResourceType(AuthorizationResourceType.ROLE)
+        .withAction(PermissionAction.REMOVE)
+        .add();
```

Comment:
❓ Why do we need to add a permission to the role? It shouldn't be required for deleting a role. Removing this also means we need to change the `limitByCount`

Reply:
It's required, because in the applier we are performing this operation on the authorization state:

```
    authorizationState.deleteAuthorizationsByOwnerKeyPrefix(roleKey);
    authorizationState.deleteOwnerTypeByKey(roleKey);
```

Reply:
I don't see why that would make adding a permission be required? `authorizationState.deleteAuthorizationsByOwnerKeyPrefix(roleKey);` is in a `whileEqualPrefix`. If there's no permissions added it will do nothing, which is fine.
`authorizationState.deleteOwnerTypeByKey(roleKey);` will work anyway as this CF is populated upon role creation. No need for a permission to be added here either.